addons/web_editor/controllers/main.py

```diff
@@ -340,12 +341,42 @@ def remove(self, ids, **kwargs):
         return removal_blocked_by
 
     @http.route('/web_editor/get_image_info', type='json', auth='user', website=True)
-    def get_image_info(self, src=''):
+    def get_image_info(self, src='', base=''):
```

Comment:
The fact that the signature of the function has been modified in a stable version is not a problem in this case. Indeed, the added parameter is non mandatory and the function works as before for a valid `src` with no `base` (which means that potential client customization that rely on this route will continue to work as expected).

Reply:
It is a problem in general to change the signature, even in this case: someone could have a custo relying on it, adding a parameter themselves. Not sure what's the best practice though: @rdeodoo would you add `**kwargs` (seems unsafe too though), or maybe we can retrieve some value via `request.env.context` automatically?

But another question may be: can't retrieve that "base" URL from the python side? Just `request.website.domain` maybe? Probably need @rdeodoo 's opinion on that too.

Reply:
I think it's fine to add named param in stable, especially at the end of the signature (removing a param is however a no go).

About the python side, indeed everything seems recoverable from the request. I guess `request.httprequest.url_root` or some property on the `httprequest` will find that info (check `headers`, `referrer` etc)

Reply:
@rdeodoo, @qsm-odoo , thanks for the explanation. In this case, I would rather go for `request.httprequest.referrer`. Indeed, if the user is in edit mode on a blog page and clicks on an image, `request.httprequest.url` is set to `http://localhost:8069/web_editor/get_image_info` in `get_image_info()` while `request.httprequest.referrer` contains the url where the requested resource is being used; `http://localhost:8069/blog/travel-1/maui-helicopter-tours-2` (this is what we want to be able to handle relative image url that does not begin with a `/`).
However, it seems to possible to change the browser referer policy. I tested on firefox and in this case, if the `network.http.referer.defaultPolicy` setting is set to `0`, we can not rely on this information. I don't know if this is a case that should be taken into account?

Reply:
I don't think `request.httprequest` is the way to go her. This relies on the request outside of the edited iframe, which may not be what you want: you want the URL base of the iframe itself.

Reply:
But the iframe base URL will always be the same as the top window URL, otherwise you are redirected to the new domain if those are different, isn't it?

And about `referrer`, if you just want the base url, isn't `url_root` enough?

Reply:
> And about referrer, if you just want the base url, isn't url_root enough?

I think `url_root` might not be enough in the case of an image that has a relative url src that does not begin with a `/`. Let's take as an example an image on `http://localhost:8069/blog/travel-1` that has a src set to `path/landscape.png`. In this case, the image is linked to an attachment of type 'url' with its url set to `/blog/path/landscape.png`. When clicking on the image and accessing `get_image_info`, `url_root` will be set to `http://localhost:8069/` which is not enough to recover the correct relative url (`/blog/path/landscape.png`).

Reply:
> But the iframe base URL will always be the same as the top window URL, otherwise you are redirected to the new domain if those are different, isn't it?

I don't think it's a good idea to rely on that, especially in stable. You can be 99% sure there is never a case where both URL do not match? I think it would be way safer to send the img's owerDocument URL to be sure, but maybe it's just because I know the JS better :shrug: 

And as @loco-odoo explained, I think he's right, url_root not being enough to handle all crappy cases once and for all. 

Did you try to send the base value via the context @loco-odoo ?

Reply:
> You can be 99% sure there is never a case where both URL do not match? 

From what I know, the iframe is loaded with a relative url, which then uses the base url of the top window, no?
Unless the two url domain (from website.domain and from top window) do not match, in which case we force a redirect to the website domain (with the popup which is telling you you are about to be redirected)

And any navigation to another domain inside the iframe seems to correctly be making the top window navigate and not the iframe.

So no, I can't say for sure it's always the case since it relies on JS code of our own, could be missing cases.

But yep if you have a 100% reliable way to get the info in js, go for it, it's clearly better. I was mainly giving the way to retrieve it from the python side as asked :+1:

Reply:
> the iframe is loaded with a relative url,

Mmh, yes, good point indeed.

Still we want the pathname of the iframe as @loco-odoo explained so we cannot rely solely on the url_root.

> I was mainly giving the way to retrieve it from the python side as asked 👍

Indeed but I missed the case @loco-odoo explained when I asked :+1: And was not sure about the context thing either but just have to test :+1: